hoot-core/src/main/cpp/hoot/core/auth/HootServicesLoginManager.cpp

```diff
@@ -0,0 +1,220 @@
+/*
+ * This file is part of Hootenanny.
+ *
+ * Hootenanny is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ *
+ * --------------------------------------------------------------------
+ *
+ * The following copyright notices are generated automatically. If you
+ * have a new notice to add, please use the format:
+ * " * @copyright Copyright ..."
+ * This will properly maintain the copyright information. DigitalGlobe
+ * copyrights will be updated automatically.
+ *
+ * @copyright Copyright (C) 2018 DigitalGlobe (http://www.digitalglobe.com/)
+ */
+
+#include "HootServicesLoginManager.h"
+
+// Hoot
+#include <hoot/core/util/Log.h>
+#include <hoot/core/util/StringUtils.h>
+#include <hoot/core/io/HootApiDb.h>
+#include <hoot/core/io/HootNetworkCookieJar.h>
+
+// Std
+#include <iostream>
+#include <string>
+
+// Boost
+#include <boost/property_tree/json_parser.hpp>
+#include <boost/foreach.hpp>
+
+namespace hoot
+{
+
+HootServicesLoginManager::HootServicesLoginManager()
+{
+}
+
+QString HootServicesLoginManager::getRequestToken(QString& authUrlStr)
+{
+  HootNetworkRequest requestTokenRequest;
+  _cookies.reset(new HootNetworkCookieJar());
+  requestTokenRequest.setCookies(_cookies);
+  try
+  {
+    LOG_VART(ConfigOptions().getHootServicesAuthRequestTokenEndpoint());
+    requestTokenRequest.networkRequest(ConfigOptions().getHootServicesAuthRequestTokenEndpoint());
+  }
+  catch (const std::exception& e)
```

Comment:
Does `HootNetworkRequest::networkRequest` throw anything other than a `HootException`?  Would that be a better exception to catch?

Reply:
It doesn't, so HootException would be better. will do that.